src/tools/rustfmt/src/parse/session.rs

```diff
@@ -40,6 +41,16 @@ impl Translate for SilentEmitter {
     fn fallback_fluent_bundle(&self) -> &rustc_errors::FluentBundle {
         panic!("silent emitter attempted to translate a diagnostic");
     }
+
+    // Override `translate_message` for the silent emitter because eager translation of
+    // subdiagnostics result in a call to this.
+    fn translate_message<'a>(
+        &'a self,
+        message: &'a rustc_errors::DiagnosticMessage,
+        _: &'a rustc_errors::translation::FluentArgs<'_>,
+    ) -> Result<Cow<'_, str>, rustc_errors::error::TranslateError<'_>> {
+        rustc_errors::emitter::silent_translate(message)
+    }
```

Comment:
Haven't got confirmation from the user yet if they're using a recent nightly version of rustfmt, but we just got a report about the `SilentEmitter` in https://github.com/rust-lang/rustfmt/issues/6082 and this PR is the only one I can think of that made changes recently. @davidtwco would you mind looking into this.

Reply:
Yes, this PR is almost certainly the cause. I assume @davidtwco will look into it soon.

Reply:
I'm working on this in #121301.

Reply:
@davidtwco thank you for the help on this one 🙏🏼